Engine/Generic/RuleInfo.cs

```diff
@@ -78,6 +80,16 @@ public RuleSeverity Severity
             private set { ruleSeverity = value; }
         }
 
+        /// <summary>
+        /// ImplementingType : The type which implements the rule.
+        /// </summary>
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode")]
```

Comment:
I think the CA rule is right to call this out. C# has auto-properties now that can simplify this property to `public Type { get; }` whereby the private setter is included by default.

Reply:
yah - I didn't want to change them all, so I just followed the previous example for consistency. I'm certainly happy to change all of them.